components/automate-cli/pkg/verifyserver/services/s3configservice/s3configservice.go

```diff
@@ -0,0 +1,147 @@
+	s3BucketAccessTitle         = "S3 bucket access test"
+	s3BucketAccessErrorMsg      = "Machine is not able to access the S3 bucket using the provided access key and secret key"
+	s3BucketAccessResolutionMsg = "Provide the necessary access to the S3 bucket"
+	s3BucketAccessSuccessMsg    = "Machine is able to access the S3 bucket using the provided access key and secret key"
+	region                      = ""
+)
+
+type IS3Config interface {
+	GetS3Connection(*models.S3ConfigRequest) *models.S3ServiceCheck
+	GetBucketAccess(*models.S3ConfigRequest) *models.S3ServiceCheck
+}
+
+type S3ConfigService struct {
+	Logger   logger.Logger
+	Req      *models.S3ConfigRequest
+	AwsUtils awsutils.AwsUtils
+}
+
+func NewS3ConfigService(logger logger.Logger, awsUtils awsutils.AwsUtils) IS3Config {
+	return &S3ConfigService{
+		Logger:   logger,
+		AwsUtils: awsUtils,
+	}
+}
+
+func (ss *S3ConfigService) GetS3Connection(req *models.S3ConfigRequest) *models.S3ServiceCheck {
+	ss.Req = req
+	region = req.Region
+	ss.Req.Region = ""
+	sess, err := ss.AwsConnection()
+	if err != nil {
+		return ss.Response(s3ConnectionTitle, "", errors.Wrap(err, s3ConnectionErrorMsg).Error(), s3ConnectionResolutionMsg, false)
+	}
+	s3Client := ss.AwsUtils.New(sess)
+	err = ss.ListBuckets(s3Client)
+	if err != nil {
+		return ss.Response(s3ConnectionTitle, "", errors.Wrap(err, s3ConnectionErrorMsg).Error(), s3ConnectionResolutionMsg, false)
+	}
+	return ss.Response(s3ConnectionTitle, s3ConnectionSuccessMsg, "", "", true)
+}
+
+func (ss *S3ConfigService) GetBucketAccess(req *models.S3ConfigRequest) *models.S3ServiceCheck {
+	ss.Req = req
+	ss.Req.Region = region
+	// S3 connection
+	sess, err := ss.AwsConnection()
+	s3Client := ss.AwsUtils.New(sess)
+	if err != nil {
+		return ss.Response(s3BucketAccessTitle, "", errors.Wrap(err, s3BucketAccessErrorMsg).Error(), s3BucketAccessResolutionMsg, false)
+	}
+
+	// upload data in s3 bucket
+	err = ss.UploadObject(sess)
+	if err != nil {
+		return ss.Response(s3BucketAccessTitle, "", errors.Wrap(err, s3BucketAccessErrorMsg).Error(), s3BucketAccessResolutionMsg, false)
+	}
+
+	// read/list data in s3 bucket
+	err = ss.ListObjects(s3Client)
+	if err != nil {
+		return ss.Response(s3BucketAccessTitle, "", errors.Wrap(err, s3BucketAccessErrorMsg).Error(), s3BucketAccessResolutionMsg, false)
+	}
+
+	// delete data in s3 bucket
+	err = ss.DeleteObjects(s3Client)
+	if err != nil {
+		return ss.Response(s3BucketAccessTitle, "", errors.Wrap(err, s3BucketAccessErrorMsg).Error(), s3BucketAccessResolutionMsg, false)
+	}
+	return ss.Response(s3BucketAccessTitle, s3BucketAccessSuccessMsg, "", "", true)
+}
+
+func (ss *S3ConfigService) AwsConnection() (*session.Session, error) {
+	sess, err := ss.AwsUtils.NewSessionWithOptions(ss.Req.Endpoint, ss.Req.AccessKey, ss.Req.SecretKey, ss.Req.Region)
+	if err != nil {
+		ss.Logger.Error("s3 config aws connection failed: ", err.Error())
```

Comment:
No unit test for this. If you could have written the tests for above mentioned scenarios, this will cover on that.